R/CreatePackageReport.R

```diff
@@ -68,52 +86,24 @@ CreatePackageReport <- function(packageName
 }
 
 
-
 # [Title] Build The Package Reporters
 # [Author] B. Burns
 # [Desc] This function creates an instance of each package reporter
 #        and enriches its content.   
 #
-#  For param descriptions, see CreatePackageReport
-
+# [seealso] For param descriptions, see CreatePackageReport
 .BuildPackageReporters <- function(packageName
                                    , packageReporters
                                    , packagePath){
-      # Input checks
-      assertthat::assert_that(
-        assertthat::is.string(packageName)
-        , is.list(packageReporters)
-      )
-      
-      # Confirm that all reporters are actually reporters
-      checks <- sapply(packageReporters, function(x){methods::is(x, "AbstractPackageReporter")})
-      if (!all(checks)){
-        msg <- paste0("At least one of the reporters in the packageReporters parameter ",
-                      "is not a PackageReporter. See ?AbstractPackageReporter for details.")
-        log_fatal(msg)
-      }
-      
-      log_info(paste0("Creating package report for package "
-                      , packageName
-                      , " with reporters:\n\n"
-                      , paste(unlist(lapply(packageReporters, function(x) class(x)[1]))
-                              , collapse = "\n")))
-      
-      
-      # Make them plots
-      for (reporter in packageReporters){
-        log_info(paste("Running Package Reporter", class(reporter)[1]))
-        reporter$set_package(packageName, packagePath)
-        
-        reporter$calculate_all_metrics()
-        
-        reporter$plot_network()
-        
-        log_info(paste("Done Package Reporter",class(reporter)[1]))
-      }
       
+      packageReporters <- sapply(
+          X = packageReporters
+          , FUN = function(reporter){
+              reporter$set_package(packageName, packagePath)
+              return(reporter)
```

Comment:
I believe you also need to run `reporter$calculate_all_metrics()` to populate each reporter.

Reply:
ok ok I take it back, you were right. Needed to add a call to actually calculate the metrics. We still do not need a public function called `calculate_all_metrics`, though. In my most recent commit, I actually implemented the thing I THOUGHT I'd done as of that last comment